crates/solver/src/driver.rs

```diff
@@ -648,6 +648,14 @@ impl Driver {
         let (mut rated_settlements, errors) = self
             .rate_settlements(solver_settlements, &external_prices, gas_price)
             .await?;
+        // We don't know the exact block because simulation can happen over multiple blocks but
+        // this is a good approximation.
+        let block_during_simulation = self
+            .block_stream
+            .borrow()
+            .number
+            .unwrap_or_default()
+            .as_u64();
```

Comment:
Not sure how accurate this value has to be but there are a few options to improve the accuracy:
1. Take the median of all simulation block numbers
2. Store a range of blocks in `SolverCompetitionResponse`
3. Store one block number per solution

Since I don't know how much a more accurate block number would help preferring the simplest solution seems sensible at the moment. 👌

Reply:
1. doesn't exist. We don't get the number during simulation. We could add getBlockNumber calls to the simulation (but that would be complex because of the code around BatchTransport) but even then it isn't documented anywhere that all the calls in one batch actually are run on the same block. 

Range of blocks is interesting so we have a bound between which simulation happened. Could still be wrong in case of reorg. Anyway, the number doesn't have to be super accurate so I prefer the current simple solution.